### ISSUE TITLE
json2csv doesn't flatten values correctly

### ISSUE THREAD

**Issue by joshuali925** [mrodrig/json-2-csv#184]
## Background Information
- Module Version: `3.11.1`
- Node/Browser Version: `10.23.1`

The issue I'm reporting is with:
- [x] json2csv
- [ ] csv2json

I have...
- [x] searched to see if an issue has already been reported.
- [x] verified that my JSON/CSV data is valid (using something like http://jsonlint.com or https://csvlint.io/).
- [x] tried upgrading to the latest version of json-2-csv (since the issue may already be fixed).

## Expected Behavior
outputs values

## Actual Behavior
outputs null

## Data Sample 
CSV:
```csv

```

(or)

JSON:
```json
{
  "a.a": "a",
  "a.b": {
    "c": "b"
  }
}
```

## Code Example
```bash
❯ cat test.json
{
  "a.a": "a",
  "a.b": {
    "c": "b"
  }
}

❯ npx json2csv test.json
a.a,a.b.c
a,null
```

**Comment by mrodrig** [comment]
Thanks for reporting this @joshuali925. This wasn't a scenario I considered, but it's definitely valid. It looks like I'll need to rework things a bit in both the underlying key path generation module (`deeks`) and key path evaluation module (`doc-path`) to handle this in an efficient manner. I'll post back when I have progress to report.

**Comment by mrodrig** [comment]
Just wanted to give a quick update - I have the needed changes in the `deeks` and `doc-path` dependency modules complete and released to NPM. The only change that was needed to `json-2-csv` is also pushed up in a branch, but I need to formalize my manual test scripts into automated unit tests before I can get it out in a release. I'm hoping to have those written up and pushed in the next couple of days. Once those are all set, I should be able to release this shortly afterwards. Thanks for your patience.

**Comment by joshuali925** [comment]
Got it, thanks a lot for your work!

**Comment by mrodrig** [comment]
I just merged and published the fix in `3.14.0` on NPM. Thanks again for reporting this. Feel free to let me know if you spot anything else that doesn't look right.
